src/serial.zig

```diff
@@ -745,6 +747,14 @@ pub const SerialConfig = struct {
 
     /// Defines the handshake protocol used.
     handshake: Handshake = .none,
+
+    pub fn bufPrint(self: Self, buf: []u8) ![]u8 {
+        return std.fmt.bufPrint(buf, "{d}@{d}{c}{d}{s}", .{ self.baud_rate, @intFromEnum(self.word_size), @intFromEnum(self.parity), @intFromEnum(self.stop_bits), switch (self.handshake) {
+            .none => "",
+            .hardware => " RTS/CTS",
+            .software => " XON/XOFF",
+        } });
+    }
```

Comment:
If you replace `bufPrint` with
```zig
pub fn format(conf: SerialConfig, fmt: []const u8, options: std.fmt.FormatOptions, writer: anytype) !void {
  …
}
```
You can just do `whatever.print("{}", .{ serialConfig })` and format it not only with bufPrint but any printing function :)

the bufPrint usage would then be:
```zig
var buffer: […]u8 = undefined;
const formatted = try std.fmt.bufPrint(&buffer, "{}", .{config});
```

Reply:
Thanks, that looks much better. Have updated